internal/graph/cursors_test.go

```diff
@@ -212,3 +214,142 @@ func TestCombineCursorsWithNil(t *testing.T) {
+}
+
+func TestWithParallelizedStreamingIterableInCursorWithExistingCursor(t *testing.T) {
+	limits, _ := newLimitTracker(context.Background(), 50)
+	revision := revision.NewFromDecimal(decimal.NewFromInt(1))
+
+	ci, err := newCursorInformation(&v1.Cursor{
+		AtRevision: revision.String(),
+		Sections:   []string{"iter", "2"},
+	}, revision, limits)
+	require.NoError(t, err)
+
+	items := []int{10, 20, 30, 40, 50}
+	parentStream := dispatch.NewCollectingDispatchStream[int](context.Background())
+	err = withParallelizedStreamingIterableInCursor[int, int](
+		context.Background(),
+		ci,
+		"iter",
+		items,
+		parentStream,
+		2,
+		func(ctx context.Context, cc cursorInformation, item int, stream dispatch.Stream[int]) error {
+			err := stream.Publish(item * 10)
+			if err != nil {
+				return err
+			}
+
+			return stream.Publish((item * 10) + 1)
+		})
+
+	require.NoError(t, err)
+	require.Equal(t, []int{300, 301, 400, 401, 500, 501}, parentStream.Results())
+}
+
+func TestWithParallelizedStreamingIterableInCursorWithLimit(t *testing.T) {
+	limits, _ := newLimitTracker(context.Background(), 5)
+	revision := revision.NewFromDecimal(decimal.NewFromInt(1))
+
+	ci, err := newCursorInformation(&v1.Cursor{
+		AtRevision: revision.String(),
+		Sections:   []string{},
+	}, revision, limits)
+	require.NoError(t, err)
+
+	items := []int{10, 20, 30, 40, 50}
+	parentStream := dispatch.NewCollectingDispatchStream[int](context.Background())
+	err = withParallelizedStreamingIterableInCursor[int, int](
+		context.Background(),
+		ci,
+		"iter",
+		items,
+		parentStream,
+		2,
+		func(ctx context.Context, cc cursorInformation, item int, stream dispatch.Stream[int]) error {
+			err := stream.Publish(item * 10)
+			if err != nil {
+				return err
+			}
+
+			return stream.Publish((item * 10) + 1)
+		})
+
+	require.NoError(t, err)
+	require.Equal(t, []int{100, 101, 200, 201, 300}, parentStream.Results())
+}
+
+func TestWithParallelizedStreamingIterableInCursorEnsureParallelism(t *testing.T) {
+	limits, _ := newLimitTracker(context.Background(), 500)
+	revision := revision.NewFromDecimal(decimal.NewFromInt(1))
+
+	ci, err := newCursorInformation(&v1.Cursor{
+		AtRevision: revision.String(),
+		Sections:   []string{},
+	}, revision, limits)
+	require.NoError(t, err)
+
+	items := []int{}
+	expected := []int{}
+	for i := 0; i < 500; i++ {
+		items = append(items, i)
+		expected = append(expected, i*10)
+	}
+
+	encountered := []int{}
+	lock := sync.Mutex{}
+
+	parentStream := dispatch.NewCollectingDispatchStream[int](context.Background())
+	err = withParallelizedStreamingIterableInCursor[int, int](
+		context.Background(),
+		ci,
+		"iter",
+		items,
+		parentStream,
+		5,
+		func(ctx context.Context, cc cursorInformation, item int, stream dispatch.Stream[int]) error {
+			lock.Lock()
+			encountered = append(encountered, item)
+			lock.Unlock()
+
+			return stream.Publish(item * 10)
+		})
+
+	require.Equal(t, len(expected), len(encountered))
```

Comment:
I think I'm missing how this tests concurrency?

Reply:
`require.NotEqual(t, encountered, expected)` does so